hail/python/hailtop/gear/auth/tokens.py

```diff
@@ -17,14 +18,17 @@ def get_tokens_file():
             return os.path.expanduser('~/.hail/tokens.json')
         return '/user-tokens/tokens.json'
 
-    def __init__(self):
-        tokens_file = self.get_tokens_file()
+    @staticmethod
+    def from_file():
+        tokens_file = Tokens.get_tokens_file()
         if os.path.isfile(tokens_file):
             with open(tokens_file, 'r') as f:
-                self._tokens = json.loads(f.read())
+                return Tokens(json.loads(f.read()))
         else:
-            log.info(f'tokens file not found: {tokens_file}')
-            self._tokens = {}
+            raise NoTokenFileFoundError(tokens_file)
```

Comment:
I think the error shouldn't be on the missing tokens file (which should be the same as an empty tokens file, so we can do the read/modify/write trick), but actually asking for a token for a namespace which is not present.